src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -921,6 +921,29 @@ public Path zipRepository(Repository repo, String targetPath, boolean hideStuden
         return zipFileService.createZipFileWithFolderContent(zipFilePath, repoPath);
     }
 
+    /**
+     * Zip the content of a git repository available for an instructor/tutor.
+     * @param repo Local Repository Object.
+     * @param zipName The name of the zipped file
+     * @param targetPath path where the repo is located on disk
+     * @return The path to the zipped file
+     * @throws IOException if the zipping process failed.
+     */
+    public Path zipInstructorRepository(Repository repo, String zipName, String targetPath) throws IOException {
```

Comment:
do we really need this? or could this be implemented with an existing method?
We should avoid code duplication as it leads to additional effort